src/plugins/data_source/server/saved_objects/credential_saved_objects_type.ts

```diff
@@ -32,6 +33,7 @@ export const credential: SavedObjectsType = {
       credentialType: { type: 'keyword', index: false },
       credentialMaterials: { type: 'object' },
       description: { type: 'text', index: false },
+      version: { type: 'integer', index: false },
```

Comment:
do we really need to define these field mappings? shall we just define `type` and `type` like index-pattern does?

Reply:
I'm wondering if we should use explicit mapping rather than dynamic mapping. Because index-pattern only defines 'title' and 'type,' we can easily send adhoc requests via saved object API.

```
curl --include -XPOST http://localhost:5603/qba/api/saved_objects/index-pattern \
    -H "Content-Type: application/json" \
    -H "osd-xsrf: true" \
    -d'{"attributes": {"title": "aaa", "type": "bbb", "adhoc": "abcd"}}'
HTTP/1.1 200 OK

{"type":"index-pattern","id":"a55a5480-13f1-11ed-ac4d-79ddf63b23ce","attributes":{"title":"aaa","type":"bbb","adhoc":"abcd"},"references":[],"migrationVersion":{"index-pattern":"7.6.0"},"updated_at":"2022-08-04T12:33:30.056Z","version":"WzQzLDFd","namespaces":["default"]}%                                                                                                       

curl --include -XPOST http://localhost:5603/qba/api/saved_objects/index-pattern \
    -H "Content-Type: application/json" \
    -H "osd-xsrf: true" \
    -d'{"attributes": {"title": "aaa", "type": "bbb", "adhoc2": "abcd"}}'
HTTP/1.1 200 OK

{"type":"index-pattern","id":"bf1146e0-13f1-11ed-ac4d-79ddf63b23ce","attributes":{"title":"aaa","type":"bbb","adhoc2":"abcd"},"references":[],"migrationVersion":{"index-pattern":"7.6.0"},"updated_at":"2022-08-04T12:34:13.198Z","version":"WzQ0LDFd","namespaces":["default"]}%
```

Reply:
you will still have dynamic mapping anyways, isn't it? on line 30 you have `dynamic: false,` which means you still accept arbitrary fields. I don't see big benefit to limit the fields, while it impact the flexibility

Reply:
As discussed offline, will keep title and authType